sycl/include/sycl/ext/oneapi/sub_group_mask.hpp

```diff
@@ -252,6 +252,8 @@ struct sub_group_mask {
   sub_group_mask(const sub_group_mask &rhs)
       : Bits(rhs.Bits), bits_num(rhs.bits_num) {}
 
+  sub_group_mask &operator=(sub_group_mask &rhs) = delete;
```

Comment:
```suggestion
  sub_group_mask &operator=(const sub_group_mask &rhs) = delete;
```
And again, why do we delete it? Seems to me like it should be valid to use copy assignment.

Reply:
Good point. Fixed. Thanks!

Reply:
Why is the copy assignment operator still getting deleted instead of providing a proper implementation? This causes issues in `Kokkos`/`desul`, see https://github.com/desul/desul/blob/d2a78e176e0a30f5893c596012d1115df5289be7/atomics/include/desul/atomics/Compare_Exchange_SYCL.hpp#L106.

Reply:
Thank you for pointing this out, @masterleinad ! Technically the copy assignment operator isn't part of [sycl_ext_oneapi_sub_group_mask](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_sub_group_mask.asciidoc) until version 2, but since we had it before I agree that we might as well have kept it around. I am adding it back in https://github.com/intel/llvm/pull/10886.